### ISSUE TITLE
custom dataset other than CIFAR/ Imagenet

### ISSUE THREAD

**Issue by gauravharjule** [Hao840/OFAKD#30]
Hello,
I was able to run your code successfully,

Now my Aim is
I want to  use the OFA loss for custom dataset, so how should I be approaching it,
Should I train the teacher from scratch or should I finetune the teacher model for the same then do it
What all Hyperparameters I should be focusing upon to do the same

Your suggestions being the author will be very helpful

**Comment by Hao840** [comment]
Hi, if you want to run on custom dataset, you should define the creation of your own dataset first.
As for the teacher model, you can either train it from scratch or finetune based on imagenet pretrained checkpoints. If the scale of your dataset is relative small, I recommand to finetune rather than train from scratch.
Default hyperparameters may work, but I think adjusting them for your dataset could achieve better performance.

**Comment by gauravharjule** [comment]
Thanks, I'll check them out.

**Comment by gauravharjule** [comment]
where will I get your check points for imagenet,
You have recently only released the CIFAR100 checkpoint s and there is only imagenet checkpoint for diet

Request if you can share the same if possible

Thanks

**Comment by Hao840** [comment]
for more imagenet checkpoints, please refer to their official repo.

**Comment by gauravharjule** [comment]
Hey thanks for that, but

If I use pre trained check points from timm library and fine tune it,  them
once I do the distillation using OFAKD there is an issue when we load the
teacher, it gives error unexpected values


That's why I requested if you can share the pretrained check  points of
imagenets you tried from scratch for various teacher models using OFAKD

Thanks and regards

**Comment by Hao840** [comment]
We just use checkpoints provided by timm on ImageNet. Maybe there are something wrone with your configuration during distillation?

**Comment by gauravharjule** [comment]
Hiee
Want to seek theoretical clarification regarding the loss function derived.

Which formula you have used for KL divergence where you neglected the
denominator term and derived the equation no 5.

Request if you can give some ref for the same,  was trying to derive from
original formula but was not able to figure it out


Regards
